ensime-company.el

```diff
@@ -125,7 +125,7 @@
   (yas-minor-mode-on)
   (set (make-local-variable 'company-idle-delay) 0)
   (set (make-local-variable 'company-minimum-prefix-length) 2)
-  (local-set-key [tab] 'ensime-company-complete-or-indent))
+  (local-set-key (kbd "TAB") 'ensime-company-complete-or-indent))
```

Comment:
The problem is that in tty the physical TAB key is bound to C-i and "TAB".
To distinguish C-i and "TAB" in the window-system, another key was introduced <tab> which doesn't translate to C-i. (freeing this combination)
A better explanation can be found in: https://www.emacswiki.org/emacs/TabKey
and of cause the docstrings of edmacro-mode:

```
 * The special words RET, SPC, TAB, DEL, LFD, ESC, and NUL represent
   special control characters.  The words must be written in uppercase.

 * A word in angle brackets, e.g., <return>, <down>, or <f1>, represents
   a function key.  (Note that in the standard configuration, the
   function key <return> and the control key RET are synonymous.)
   You can use angle brackets on the words RET, SPC, etc., but they
   are not required there.
``` 
So, you are right, if somebody did bind a command to C-i, this fix could potentially break the system. (Perhaps Evil mode does such bindings,.....?)

Could this be corrected with a (if (window-system) ...)
What do you think?